### ISSUE TITLE
Allow HTTP Icon change not performed on task bar icon

### ISSUE THREAD

**Issue by GoogleCodeExporter** [soul4soul/peerblock#176]
```
What steps will reproduce the problem?
1. Run PB r202
2. Allow HTTP.  PB's window icon and system tray icons will change to yellow.
3.

What is the expected output? What do you see instead?
The task bar icon stays blue.

What version of PeerBlock are you using? On what operating system? 32- or
64-bit?
Windows 7 64bit.


Please provide any additional information below.  Make sure to attach
peerblock.log and/or any screenshots that would help explain your problem.


```

Original issue reported on code.google.com by `saki...@gmail.com` on 29 Oct 2009 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
Yes, this is the new behavior of that icon - it changes yellow to warn you that 
you
have HTTP Allowed and are currently not fully protected.  Don't forget, by "HTTP
Allowed" what we really mean is "Allow any traffic going out port 80 or 443 to 
pass
through, regardless of what it looks like or where it's going".  So if, say, 
someone
were to setup their bittorrent program to only use port 80, they would be able 
to
connect to you even if they're on one of your blocklists.  

And I have recently seen an outgoing connection attempt on port 80 to an IP 
address
flagged as belonging to Mediasentry...  Dangerous!

If you're asking for an option to ignore this warning icon color change and 
stay blue
all the time we will certainly consider it, but we do strongly recommend 
blocking
HTTP on a general basis for the above reasons.  I realize that blocking HTTP 
can be
pretty painful at times, and that's partially why we implemented this yellow 
icon
change - when you're surfing the web and are willing to temporarily lower your
computer's protection level you can Allow HTTP, and it will turn yellow to 
remind you
to Block HTTP again once you're finished.
```

Original comment by `peerbloc...@gmail.com` on 30 Oct 2009 at 12:49

**Comment by GoogleCodeExporter** [comment]
```
Howdy.
What I am saying is that when this new feature is used, that simply the PB icon 
in
the task bar is not changed to yellow when the PB window is open or minimized.  
PB's
window icon and system tray icon change to yellow correctly.
```

Original comment by `saki...@gmail.com` on 30 Oct 2009 at 1:31

**Comment by GoogleCodeExporter** [comment]
```
FTR , I exited 198 , installed 202 , started 202 , and tested in XP Pro SP2 :

(blue / yellow / red) colors are working OK in the 3 mentioned places :

systray icon , window title line , and task bar .
```

Original comment by `JRF...@gmail.com` on 30 Oct 2009 at 11:58

**Comment by GoogleCodeExporter** [comment]
```
Sorry for misinterpreting your original bug, sakison.  I do see this problem on 
Win 7
x64 as well, but not on Vista x64 - looks like it may be a difference in the 
way the
Win 7 taskbar operates, since it looks like it works on XP too as per JRF999's 
post.

Marking as low priority since the program is working just fine, it's just a 
display
issue.  Targetting for 1.1 however since it would be nice to have this resolved 
in
there prior to releasing it as a "Stable"/Final release.
```

Original comment by `peerbloc...@gmail.com` on 30 Oct 2009 at 1:43
- Changed state: **Accepted**
- Added labels: **Milestone-1.1**, **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
This appears to be related to the Windows 7 taskbar putting an "admin" overlay 
onto
the PeerBlock icon there.  When that overlay is present, the taskbar icon 
doesn't
change; when that overlay is NOT present, the icon DOES change colors.  
Sometimes the
overlay is present, other times it's not, and I haven't yet discerned the 
pattern
behind it...
```

Original comment by `peerbloc...@gmail.com` on 19 Nov 2009 at 5:05

**Comment by GoogleCodeExporter** [comment]
```
Also, it appears as though if you've "pinned" PeerBlock to your taskbar, 
Windows 7
may be caching the icon for it so any change to the icon won't show up until you
logoff/login again, or switch users (?), or something. 
(http://social.technet.microsoft.com/Forums/en/w7itprosecurity/thread/bc8f8d98-f
b31-456a-bc05-11576acbe65b)
```

Original comment by `peerbloc...@gmail.com` on 19 Nov 2009 at 5:56

**Comment by GoogleCodeExporter** [comment]
```
Looks like this is in fact an issue caused by some "quirks" of the new Windows 
7 taskbar.

Firstly, if an app is "pinned" to the taskbar, we're not allowed to change its 
icon.
 This is a pain, and many devs appear to have complained to MS about this, but it is
what it is at this point.

There also appears to be a slight issue with even non-pinned apps, at least on 
Win7
RC1 which I'm running - sometimes Windows gets confused and applies the above 
"icon
is locked" rules as if we were pinned.  If this is the case you'll see a little 
"UAC
Shield" in the lower-right corner of your PeerBlock taskbar icon.  To get rid 
of this
and get the icon to change again (assuming you're not pinned), you'll need to 
Pin the
app and then Un-pin it again.

Not sure whether we'll ever be able to patch around this behavior - this is 
primarily
a Microsoft issue.  The best we can likely do is add an "overlay" icon, 
basically our
own little icon to the lower-right of the main icon showing PeerBlock's status. 
Won't be nearly as noticeable as a full main-icon color-change, but it's 
probably the
best we can do.
```

Original comment by `peerbloc...@gmail.com` on 1 Dec 2009 at 3:17

**Comment by GoogleCodeExporter** [comment]
```
Not going to get to this in the 1.1 timeframe, so marking it as targetted for 
1.2
instead.
```

Original comment by `peerbloc...@gmail.com` on 30 Mar 2010 at 4:28
- Added labels: **Milestone-1.2**
- Removed labels: **Milestone-1.1**
